### PR TITLE
[wip] Add support for pod inside of user namespace.

### PR DESCRIPTION
**Pull request by rhatdan**
Specifically we need to be able to launch a pod with
--userns=keep-id

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixes: https://github.com/containers/podman/issues/7706

**Comment by rhatdan** [comment]
@umohnani8 Could you take this over?  I don't have the time to complete it.

**Comment by rhatdan** [comment]
@umohnani8 Do you have a chance to take this over?

**Comment by rhatdan** [comment]
The basic issue, we are seeing is people adding containers to pods where they use additional fields that do not match the pods infra container.

For example we have an issue where a user creates a pod, and then adds a container to the pod, within a user namespace, and the /dev/mqueue from the infra container fails to mount on the container, because the /dev/mqueue inside of the pod does not match the usernamespace inside of the container.

Bottom line, I believe we have to be able to at least configure the usernamespace of the pod.

**Comment by umohnani8** [comment]
Will be taking over this work.

**Comment by umohnani8** [comment]
Replaced by https://github.com/containers/podman/pull/10589
